src/shims/mod.rs

```diff
@@ -27,9 +27,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         // There are some more lang items we want to hook that CTFE does not hook (yet).
         if this.tcx.lang_items().align_offset_fn() == Some(instance.def.def_id()) {
-            // FIXME: return a real value in case the target allocation has an
-            // alignment bigger than the one requested.
-            let n = u128::max_value();
+
+            let n = {
+                let ptr = this.force_ptr(this.read_scalar(args[0])?.not_undef()?)?;
+                let align = this.force_bits(
+                    this.read_scalar(args[1])?.not_undef()?,
+                    this.pointer_size()
+                )? as usize;
+
+                let stride = this.memory().get(ptr.alloc_id)?.align.bytes() as usize;
```

Comment:
Why do you call this "stride"...? It is the allocation's guaranteed alignment, stride is a term coming up during array layout. I am very confused.